### PR TITLE
Added support for optoanalysis

### PR DESCRIPTION
**Pull request by meganbkratz**
-- added ability to specify different pre- and post-synaptic cell groups
-- change group_results function so that a pair can be included in the aggregate for multiple pair_groups if it fits in multiple pair_groups
-- change cell:ext_id and various job_ids to be str instead of float
-- added expt_uid to experiment table

### CODE REVIEW

File: multipatch_analysis/config.py [hunk 1]
```diff
@@ -73,7 +73,7 @@
 if not os.path.isfile(configfile):
     open(configfile, 'wb').write(template)
 
-config = yaml.load(open(configfile, 'rb'))
```
Comment (campagnola) [inline multipatch_analysis/config.py]: Thanks :D
I have been ignoring this warning for some time..

File: multipatch_analysis/database/experiment.py [hunk 2]
```diff
@@ -84,6 +88,7 @@ def __repr__(self):
         ('target_temperature', 'float', 'The intended temperature of the experiment (but actual recording temperature is stored elsewhere)'),
         ('date', 'datetime', 'The date of this experiment'),
         ('lims_specimen_id', 'int', 'ID of LIMS "CellCluster" specimen.'),
+        ('expt_uid', 'str', 'Unique external identifier string for the experiment.')
```
Comment (campagnola) [inline multipatch_analysis/database/experiment.py]: Let's make this `ext_id` to stay consistent with other tables.

**Comment by campagnola** [comment]
Looks really good to me! I made one minor edit.
The only thing we can't merge right now is the change to the cell/electrode schemas, since I'll need to build a whole new DB for those (and probably deal with a few downstream effects). I suggest taking those out of this PR for now so that we can get everything else merged in.

File: multipatch_analysis/matrix_analyzer/matrix_display.py [hunk 3]
```diff
@@ -196,13 +196,28 @@ def update_matrix_display(self, results, group_results, cell_classes, cell_group
         bordercolor.fill(default_bordercolor)
         self.matrix_display_filter.colormap_legend()
 
+        ## for backwards compatability and to avoid an api change, make it okay to just pass cell_classes
+        if pre_cell_classes == None:
+            pre_cell_classes = cell_classes
+        if post_cell_classes == None:
+            post_cell_classes = cell_classes
```
[the remark this answers is not in the document]
Comment (meganbkratz) [reply multipatch_analysis/matrix_analyzer/matrix_display.py]: It depends where else this function gets called. If this is the only place, then no, we don't need to keep passing self.cell_classes.

[the remark this answers is not in the document]
Comment (scseeman) [reply multipatch_analysis/matrix_analyzer/matrix_display.py]: Yeah that's the only place it get's called, and I'd be happy to force it into the pre_classes and post_classes for future additions if this were to get called somewhere else

[the remark this answers is not in the document]
Comment (meganbkratz) [reply multipatch_analysis/matrix_analyzer/matrix_display.py]: Okay, I'll change that.

**Comment by meganbkratz** [comment]
I'm closing this PR, because I just submitted another, minus the db schema changes.

**Pull request closed without merging**